libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,370 @@
+import { classMap } from 'lit/directives/class-map.js';
+import { registerWebComponents } from '@domg-wc/common-utilities';
+import { VlButtonInputAddon, VlIconElement } from '@domg-wc/elements';
+import { live } from 'lit/directives/live.js';
+
+export const DatepickerDefaults = {
+    ...FormControlDefaults,
+    block: false,
+    readonly: false,
+    type: '' as 'range' | 'time' | 'date-time',
+    value: '',
+    format: 'd.m.Y',
+    selectedDate: '',
+    minDate: '',
+    maxDate: '',
+    minTime: '',
+    maxTime: '',
+    amPm: '',
+    pattern: '',
+};
+
+@customElement('vl-datepicker-next')
+export class VlDatepickerComponent extends FormControl {
+    // Properties
+    private value = DatepickerDefaults.value;
+    private type = DatepickerDefaults.type;
+    private block = DatepickerDefaults.block;
+    private readonly = DatepickerDefaults.readonly;
+    private format = DatepickerDefaults.format;
+    private selectedDate = DatepickerDefaults.selectedDate;
+    private minDate = DatepickerDefaults.minDate;
+    private maxDate = DatepickerDefaults.maxDate;
+    private minTime = DatepickerDefaults.minTime;
+    private maxTime = DatepickerDefaults.maxTime;
+    private amPm = DatepickerDefaults.amPm;
+    private pattern = DatepickerDefaults.pattern;
+
+    // Variables
+    instance?: Instance;
+    private initialValue = '';
+
+    static {
+        registerWebComponents([VlButtonInputAddon, VlIconElement]);
+    }
+
+    static get properties() {
+        return {
+            value: { type: String, reflect: true },
+            type: { type: String },
+            format: { type: String },
+            selectedDate: { type: String, attribute: 'selected-date' },
+            minDate: { type: String, attribute: 'min-date' },
+            maxDate: { type: String, attribute: 'max-date' },
+            minTime: { type: String, attribute: 'min-time' },
+            maxTime: { type: String, attribute: 'max-time' },
+            amPm: { type: String, attribute: 'am-pm' },
+            block: { type: Boolean },
+            readonly: { type: Boolean },
+            error: { type: Boolean },
+            success: { type: Boolean },
+            pattern: { type: String },
+        };
+    }
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [
+            resetStyle,
+            baseStyle,
+            iconStyle,
+            inputFieldStyle,
+            inputAddonStyle,
+            inputGroupStyle,
+            tooltipStyle,
+            datepickerStyle,
+            datepickerUigStyle,
+        ];
+    }
+
+    constructor() {
+        super();
+
+        // Fix tot <-> tot-en-met
+        if (Dutch?.nl) {
+            Dutch.nl.rangeSeparator = ' tot en met ';
+        }
+    }
+
+    protected firstUpdated(changedProperties: Map<string, unknown>): void {
+        super.firstUpdated(changedProperties);
+
+        this.initializeComponent();
+
+        if (!this.isInShadowDOM()) {
```

Comment:
Gaan we niet altijd gebruik maken van een Shadow DOM?

Reply:
Verwijderd.